src/haxesharp/collections/Dictionary.hx

```diff
@@ -134,29 +264,73 @@ abstract Dictionary<K,V>(BalancedTree<K,V>)
 
         for (key in allKeys)
         {
-            remove(key);
+            this.remove(key);
         }
     }
 
     /**
     Returns the number of elements in the Dictionary
-    Note that this is an O(N) operation
+    Note that this is an O(N) operation.
     */
     public function count()
     {
-        var count = 0;
-        for (value in this)
-        {
-            count++;
-        }
-        return count;
+        return this.keys().length;
     }
 
-    private inline function handleNullKey(key:K)
+    /**
+    Given a key type that's only available as a compile-time constant, determine
+    what it is at runtime. We use a combination of Type.getType and Type.getClass
+    to figure this out.
+    
+    The actual values are probably internal Haxe implementation details, and may
+    change from target to target or at runtime.
+    **/
+    private inline function definitivelyDetermineType(key:K):InternalKeyType
     {
         if (key == null)
         {
-            throw new ArgumentNullException("key was null");
+            throw new ArgumentNullException("key");
+        }
+
+        var typeOf = Type.typeof(key).getName();
```

Comment:
Is there any reason we throw away the `ValueType` enum value that `typeof(key)` gives us and get the stringified version of it instead? It seems to me like the `ValueType` would actually be more useful, as it *should* make this code more resistant if anything in Haxe's Type API ever changes.  This might be a good use case for a `switch` statement as well.